jmeter-java-dsl/src/main/java/us/abstracta/jmeter/javadsl/core/listeners/autostop/AutoStopTestBean.java

```diff
@@ -79,28 +79,42 @@ public boolean isStats() {
   @Override
   public synchronized void sampleOccurred(SampleEvent e) {
     for (AutoStopConditionElement condition : conditions) {
-      if (condition.getRegex() == null && regex != null
-          && !regex.matcher(e.getResult().getSampleLabel()).matches()) {
+      if (shouldSkipCondition(e, condition)) {
+        continue;
+      }
+      if (evaluateAndStop(e, condition)) {
```

Comment:
What about something like this?:

```java
public synchronized void sampleOccurred(SampleEvent e) {
    for (AutoStopConditionElement condition : conditions) {
  /*
  Since stop is async, we check that result property is not already set to avoid overriding it
  and losing the original condition that triggered the stop
  */
  if (stopped) {
    return;
  }
  for (AutoStopConditionElement condition: conditions) {
    if (shouldSkipCondition(condition))) {
      continue;
    }
    if (condition.eval(e.getResult)) {
      stopped = true; // keeping it here, instead of adding to stop test, to keep usages of variable stop near (more cohesive code)
      stopTest(condition);
      break;
    }
  }
```